.github/workflows/publish-release.yml

```
@@ -0,0 +1,47 @@
+name: Publish to PyPI
+
+on:
+  release:
+    types: [published]
+
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  Release:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.12"]
+    permissions:
+      id-token: write
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Get release tag
+        id: get_tag
+        run: echo "RELEASE_TAG=${GITHUB_REF#refs/tags/}" >> $GITHUB_OUTPUT
+      - name: Update version
+        run: |
+          sed -i "s/version = .*/version = \"${{ steps.get_tag.outputs.RELEASE_TAG }}\"/" pyproject.toml
```

I simplified this based on what I had in other release workflows
```suggestion
      - name: Update version
        run: |
          export RELEASE_VERSION=$(echo ${{ github.ref_name }} | sed -e 's/v//g')
          sed -i "s/version = .*/version = \"$RELEASE_VERSION\"/" pyproject.toml
```